clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/AbstractVeniceAggVersionedStats.java

```diff
@@ -50,7 +50,7 @@ public AbstractVeniceAggVersionedStats(
     loadAllStats();
   }
 
-  public final synchronized void loadAllStats() {
+  public synchronized void loadAllStats() {
     metadataRepository.getAllStores().forEach(store -> {
       addStore(store.getName());
       updateStatsVersionInfo(store.getName(), store.getVersions(), store.getCurrentVersion());
```

Comment:
So, a higher level question. Should every SN register metrics for every store? Can we instead move the improvement of registering only the necessary metrics one-level up so everything benefits?

Reply:
I think we should do something like that, but I think in order to do that we have to look at every implementer of this class and figure out what the policy should be.

Probably what we could do is add an overridable method which supports filtering of a store, and have them implement that.  But I think doing that dive is out of scope for this PR.

Reply:
Yeah, I agree about it being out of scope